src/mono/mono/metadata/threads.c

```diff
@@ -1242,8 +1242,7 @@ start_wrapper_internal (StartInfo *start_info, gsize *stack_ptr)
 
 	if (G_UNLIKELY (external_eventloop)) {
 		/* if the thread wants to stay alive in an external eventloop, don't clean up after it */
-		if (mono_thread_platform_external_eventloop_keepalive_check ())
```

Comment:
You don't need support for escaping the thread start function anymore? How do the JS event loop workers start up? Are they required to use an API that exits out to the event loop on its own?

Reply:
all `mono_thread_platform_external_eventloop_keepalive_check` does is to tell you that the JS has pending `keepAlive` counter. The real magic is throwing "unwind" which could be done with `emscripten_exit_with_live_runtime` or in JS.

We do it in JS for deputy thread and IO thread, but I don't see if for JSWebWorker right now. I will have look again.

Reply:
`JSWebWorker`  -> `mono_wasm_install_js_worker_interop` -> `Module.runtimeKeepalivePush` -> keeps living